addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,207 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             $carriers.filter(':checked').click();
         }
 
-        // Asynchronously retrieve every carrier price
-        _.each($carriers, function (carrierInput, k) {
+        self._getCurrentLocation();
+
+        _.each($carriers, async function (carrierInput, k) {
             self._showLoading($(carrierInput));
-            self._rpc({
+            const result = await self._rpc({
                 route: '/shop/carrier_rate_shipment',
                 params: {
                     'carrier_id': carrierInput.value,
                 },
-            }).then(self._handleCarrierUpdateResultBadge.bind(self));
+            })
+            self._handleCarrierUpdateResult(result)
         });
-
         return this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @private
+     */
+    _getCurrentLocation: async function () {
+        const self=this;
+        const data = await self._rpc({
+            route: "/shop/access_point/get",
+        })
+        const order_locations = document.getElementsByClassName("o_order_location");
+        for (const order_loc of order_locations) {
+            const show_loc = order_loc.parentElement.nextElementSibling;
+            // We could end not having the property
+            let delivery_type=""
+            if(order_loc.parentElement.parentElement.parentElement.children[0].attributes.getNamedItem("delivery_type")){
```

Comment:
Use `closest`, then `querySelector` instead of using the parent multiple times. We can now use optional chaining to make sure we don't cause any errors.
```
order_loc.closest("parent_selector")?.querySelector("child selector")?.attributes..
```